sdk/servicebus/service-bus/src/core/batchingReceiver.ts

```diff
@@ -236,114 +400,62 @@ export class BatchingReceiver extends MessageReceiver {
         }
       };
 
       // Action to be performed on the "receiver_drained" event.
       const onReceiveDrain: OnAmqpEvent = () => {
-        if (this._receiver) {
-          this._receiver.removeListener(ReceiverEvents.receiverDrained, onReceiveDrain);
-          this._receiver.drain = false;
-        }
+        receiver.removeListener(ReceiverEvents.receiverDrained, onReceiveDrain);
+        receiver.drain = false;
 
         log.batching(
-          "[%s] Receiver '%s' drained. Resolving receiveMessages() with %d messages.",
-          this._context.namespace.connectionId,
-          this.name,
-          brokeredMessages.length
+          `${loggingPrefix} Drained, resolving receiveMessages() with ${brokeredMessages.length} messages.`
         );
 
         resolve(brokeredMessages);
       };
 
-      const onReceiveClose: OnAmqpEventAsPromise = async (context: EventContext) => {
-        try {
-          const receiverError = context.receiver && context.receiver.error;
-          if (receiverError) {
-            log.error(
-              "[%s] 'receiver_close' event occurred. The associated error is: %O",
-              this._context.namespace.connectionId,
-              receiverError
-            );
-          }
-        } catch (err) {
-          log.error(
-            "[%s] Receiver '%s' error in onClose handler:\n%O",
-            this._context.namespace.connectionId,
-            this.name,
-            translate(err)
-          );
-        }
-      };
-
-      // Action to be taken when an error is received.
-      const onReceiveError: OnAmqpEvent = (context: Pick<EventContext, "receiver">) => {
-        const receiver = this._receiver || context.receiver!;
-        cleanupBeforeResolveOrReject(receiver, "removeDrainHandler");
-
-        const receiverError = context.receiver && context.receiver.error;
-        let error: Error | MessagingError;
-        if (receiverError) {
-          error = translate(receiverError);
-          log.error(
-            "[%s] Receiver '%s' received an error:\n%O",
-            this._context.namespace.connectionId,
-            this.name,
-            error
-          );
-        } else {
-          error = new MessagingError("An error occurred while receiving messages.");
-        }
-        reject(error);
-      };
-
       cleanupBeforeResolveOrReject = (
-        receiver: Receiver | undefined,
         shouldRemoveDrain:
           | "removeDrainHandler" // remove drain handler (not waiting or initiating a drain)
           | "leaveDrainHandler" // listener for drain is removed when it is determined we dont need to drain or when drain is completed
       ): void => {
         if (receiver != null) {
-          receiver.removeListener(ReceiverEvents.receiverError, onReceiveError);
+          receiver.removeListener(ReceiverEvents.receiverError, onError);
           receiver.removeListener(ReceiverEvents.message, onReceiveMessage);
```

Comment:
This probably isn't a new thing, but you're removing the message handler before draining right? If so...then we won't know about messages that come in while draining. Or am I misunderstanding the order of operations here?

Reply:
I believe this is how it worked previously.

It is a good point though. I think the window of time is small but it's not zero either.

I'll add this note to an existing issue I already have to look into our drain behavior (#6349)